annubes/tasks/task.py

```diff
@@ -129,15 +145,21 @@ def generate_trials(
 
         # Setup phases of trial
         self._phases = {}
-        self._phases["iti"] = np.where(self.time <= self.iti)[0]
-        self._phases["fix_time"] = np.where((self.time > self.iti) & (self.time <= self.iti+ self.fix_time))[0]
-        self._phases["input"] = np.where(self.time > self.iti + self.fix_time)[0]
+        self._phases["iti"] = np.where(self.time <= self.settings.iti)[0]
+        self._phases["fix_time"] = np.where(
+            (self.time > self.settings.iti) & (self.time <= self.settings.iti + self.settings.fix_time),
+        )[0]
+        self._phases["input"] = np.where(self.time > self.settings.iti + self.settings.fix_time)[0]
         self._choice = (self._modality_seq != "catch").astype(np.int_)
 
+        # Generate inputs and outputs
+        self._inputs = self._build_trials_inputs()
+        self._outputs = self._build_trials_outputs()
+
         # Store and return trial data
         trials = vars(self)
-        trials["inputs"] = self._build_trials_inputs()
-        trials["outputs"] = self._build_trials_outputs()
+        trials["inputs"] = self._inputs
+        trials["outputs"] = self._outputs
```

Comment:
These are now covered by `trials = vars(self)`, as they are already made attributes in lines 156-157.

Reply:
yep, but I noticed that they are needed for plotting the trials in `plot_trials`. Before it was fine because `trials` was an attribute but now it's not the case anymore